File: api/laravel10.x/app/Adapters/Controllers/UserController.php
<?php

namespace App\Adapters\Controllers;

use App\Domain\Entities\User;
use App\Adapters\Controllers\Requests\CreateUserRequest;
use App\Adapters\Controllers\Responses\UserResponse;
use App\Applications\UseCases\CreateUserUseCase;
use App\Applications\UseCases\GetAllUsersUseCase;
use Illuminate\Http\Response;

class UserController extends Controller
{
    private CreateUserUseCase $createUserUseCase;
    private GetAllUsersUseCase $getAllUsersUseCase;

    public function __construct(
        CreateUserUseCase $createUserUseCase,
        GetAllUsersUseCase $getAllUsersUseCase
    ) {
        $this->createUserUseCase = $createUserUseCase;
        $this->getAllUsersUseCase = $getAllUsersUseCase;
    }

    public function create(CreateUserRequest $request)
    {
        $user = $this->createUserUseCase->execute(
            new User($request->get('name'), $request->get('email'))
        );

        $userResponse = new UserResponse($user);
        return response()->json($userResponse, Response::HTTP_CREATED);
    }

    public function getAll()
    {
        $users = $this->getAllUsersUseCase->execute();
        return response()->json(UserResponse::collection($users), Response::HTTP_OK);
    }
}

File: api/springboot/src/main/java/com/example/demo/adapters/controllers/UserController.java
package com.example.demo.adapters.controllers;

import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

import com.example.demo.adapters.controllers.requests.CreateUserRequest;
import com.example.demo.adapters.controllers.responses.UserResponse;
import com.example.demo.application.usecases.UserUseCase;
import com.example.demo.domain.model.User;

@RestController
@RequestMapping("/api/v1/users")
public class UserController {

  private final UserUseCase userUseCase;

  public UserController(UserUseCase userUseCase) {
    this.userUseCase = userUseCase;
  }

  @PostMapping
  public ResponseEntity<UserResponse> createUser(@RequestBody CreateUserRequest request) {
    User user = new User(request.getName(), request.getEmail());
    User createdUser = userUseCase.createUser(user);
    UserResponse response = new UserResponse(createdUser.getName(), createdUser.getEmail());
    return new ResponseEntity<>(response, HttpStatus.CREATED);
  }
}
